casper/src/test/scala/coop/rchain/casper/ValidateTest.scala

```diff
@@ -338,4 +350,39 @@ class ValidateTest extends FlatSpec with Matchers with BeforeAndAfterEach with B
       Left(JustificationRegression))
     log.warns.size should be(1)
   }
+
+  "Bonds cache validation" should "succeed on a valid block and fail on modified bonds" in {
+    val (_, validators)   = (1 to 4).map(_ => Ed25519.newKeyPair).unzip
+    val bonds             = validators.zipWithIndex.map { case (v, i) => v -> (2 * i + 1) }.toMap
+    val initial           = Genesis.withoutContracts(bonds = bonds, version = 0L, timestamp = 0L)
+    val storageDirectory  = Files.createTempDirectory(s"hash-set-casper-test-genesis")
+    val storageSize: Long = 1024L * 1024
+    val activeRuntime     = Runtime.create(storageDirectory, storageSize)
+    val runtimeManager    = RuntimeManager.fromRuntime(activeRuntime)
+    val initStateHash     = runtimeManager.initStateHash
+
+    val validatorsString = validators.zipWithIndex
```

Comment:
I would prefer if the validator code were generated using your ProofOfStake class so that it gets tested too.